libs/platform/experience/carousel-navigation/carousel-navigation.component.ts

```diff
@@ -0,0 +1,410 @@
+    const scrollLeft = this.hostElement.scrollLeft * (isRtl ? -1 : 1);
+    const scrollWidth = this.hostElement.scrollWidth;
+
+    this.indicatorElements.forEach((indicator, index) => {
+      const slideStart = clientWidth * index;
+      const distance = scrollLeft - slideStart;
+      const percentage = distance / this.hostElement.clientWidth;
+      let opacity = percentage >= 0 ? 1 - percentage : 1 + percentage;
+      if (clientWidth + scrollLeft >= scrollWidth) {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        if (index === this.indicatorElements!.length - 1) {
+          // we could consider making the previous slide indicator to 0
+          // (opacity = index === this.indicatorElements!.length - 1 ? 1 : 0;)
+          opacity = 1;
+        }
+      }
+      indicator.style.setProperty('--opacity', `${opacity}`);
+    });
+  }
+
+  /**
+   * Updates the state of the carousel arrows based on the current scroll position. The method
+   * is called on scroll and on resize, to ensure that the state is always up to date.
+   */
+  protected updateArrowState(): void {
+    if (!this.items.length || !this.showArrows) return;
+    const isRtl = window.getComputedStyle(this).direction === 'rtl';
+
+    const scrollLeft = this.hostElement.scrollLeft * (isRtl ? -1 : 1);
+    const scrollWidth = this.hostElement.scrollWidth;
+    const firstItemWidth = this.items?.[0].getBoundingClientRect().width;
+    const lastItemWidth =
+      this.items?.[this.items.length - 1].getBoundingClientRect().width;
+
+    this.toggleAttribute('has-previous', scrollLeft > firstItemWidth);
+
+    this.toggleAttribute(
+      'has-next',
+      scrollLeft + this.hostElement.getBoundingClientRect().width <
+        scrollWidth - lastItemWidth
+    );
+  }
+
+  /**
+   * Calculate and update indicators for the carousel based on its current state.
+   */
+  protected setSlides(): void {
+    if (!this.items.length) return;
+
+    const gap = parseFloat(
+      window.getComputedStyle(this.hostElement).getPropertyValue('column-gap') // || '0'
+    );
+
+    const slideCount = Math.ceil(
+      (this.hostElement.scrollWidth + gap) /
+        (this.hostElement.clientWidth + gap)
+    );
+
+    if (slideCount === this.slides.length) return;
+
+    if (!slideCount || slideCount === 1) {
+      if (this.slides.length > 0) this.slides = [];
+      return;
+    }
+
+    this.slides = [...Array(slideCount).keys()].map((i) => {
+      const firstElementInSlide = this.items.findIndex((el) => {
+        return (
+          this.hostElement.scrollLeft + el.getBoundingClientRect().left >=
+          i * this.hostElement.clientWidth
+        );
+      });
+      return { index: firstElementInSlide };
+    });
+  }
+
+  /**
+   * @override Renders the navigation arrow buttons (previous/next) and indicators.
+   */
+  protected override render(): TemplateResult[] {
+    const results: TemplateResult[] = [];
+
+    if (this.showArrows) {
```

Comment:
I'm not sure that pushing template's to the array is a right and performanced way. Why not to use `when` directive?

Reply:
I do not expect performance issue, but I'll keep it aligned with all other render methods we implement.